addons/account/models/account_tax.py

```diff
@@ -206,23 +209,18 @@ def _compute_tax_group_id(self):
                 ('country_id', '=', False),
             ], limit=1)
 
-    @api.model
-    def default_get(self, fields_list):
-        # company_id is added so that we are sure to fetch a default value from it to use in repartition lines, below
-        rslt = super(AccountTax, self).default_get(fields_list + ['company_id'])
-
-        company_id = rslt.get('company_id')
-
-        repartition = rslt.setdefault('repartition_line_ids', [])
-        if 'repartition_line_ids' in fields_list and not repartition:
-            repartition.extend([
-                Command.create({'document_type': 'invoice', 'repartition_type': 'base', 'tag_ids': [], 'company_id': company_id}),
-                Command.create({'document_type': 'invoice', 'repartition_type': 'tax', 'tag_ids': [], 'company_id': company_id}),
-                Command.create({'document_type': 'refund', 'repartition_type': 'base', 'tag_ids': [], 'company_id': company_id}),
-                Command.create({'document_type': 'refund', 'repartition_type': 'tax', 'tag_ids': [], 'company_id': company_id}),
-            ])
-
-        return rslt
+    @api.depends('company_id')
+    def _compute_repartition_line_ids(self):
+        for tax in self:
+            if not tax.repartition_line_ids:
```

Comment:
Won't you have a warning if you have a `tax.repartition_line_ids`? (I've been told I always need to set a else clause in a compute)

Reply:
People lied to you[^1].
You only **need** to set a value when it is a non stored field.
Stored fields will get the value from the database if the compute function didn't change them.

[^1]: it was true for only a few months

Reply:
tags are country specific so maybe you should depends of `country_code` and reset them all the time

Reply:
Btw you compute is weird... all one2manys are supposed to be computed by `_compute_repartition_line_ids` but it's not :sweat: It looks like a hack...

Reply:
@smetl Does it look more logical now?

Reply:
@smetl 

> tags are country specific so maybe you should depend of country_code and reset them all the time

Yes but you can now put tags from other countries on any tax to impact several tax report at the same time.
Also, you probably don't want to reset the values you already input because you are changing the country.